app/views/shared/_message_controls.html.haml

```diff
@@ -1,28 +1,31 @@
 = simple_form_for message, remote: true, authenticity_token: true  do |f|
-  = f.input :claim_id, value: message.claim.id, as: :hidden
-  - if current_user.persona.is_a?(ExternalUser) && Claim::BaseClaim::VALID_STATES_FOR_REDETERMINATION.include?(message.claim.state)
+  = f.input :claim_id, value: @claim.id, as: :hidden
+
+  - if current_user.persona.is_a?(ExternalUser) && @claim.redeterminable?
```

Comment:
I would refactor current_user.persona.is_a?(ExternalUser) to a single method current_user_is_external_user?